### PR TITLE
[Bugfix] [NPU] Qwen3.5 with quantization fix

### PR DESCRIPTION
**Pull request by OrangeRedeng**
## Motivation

Fix of https://github.com/sgl-project/sglang/issues/21676.

After updating the qwen3.5 modeling and changing ```in_proj_qkv``` with ```in_proj_z``` to fused ```in_proj_qkvz``` and  ```in_proj_b``` with ```in_proj_a``` to fused ```in_proj_ba```, quantization no longer works

image

## Modifications

- [x] Fix mapping for ```in_proj_qkvz``` and ```in_proj_ba```
- [x] Refactor get_linear_scheme() method according to the get_moe_scheme()
- [x] Add TODO comments. (We plan to get rid of ```packed_module_mapping``` in ```model_loader.py```, and completely switch to ```packed_module_mapping``` declared in model files, which will make the code more logical and eliminate such problems in the future)

## Accuracy Tests
Server:

```STREAMS_PER_DEVICE=32 HCCL_OP_EXPANSION_MODE="AIV" AUTO_USE_UC_MEMORY=0 P2P_HCCL_BUFFSIZE=20 python -m sglang.launch_server --device npu --attention-backend ascend --trust-remote-code --tp-size 4 --model-path ./quant_models/Qwen3_5_35B_w8a8_test_with_attn_qkv/ --port 30088 --cuda-graph-max-bs 64 --mem-fraction-static 0.8```

Client:

```python ./benchmark/gsm8k/bench_sglang.py --num-questions  1319 --port 30088 --data-path /mnt/share/tbaydasov/grade-school-math/grade_school_math/data/test.jsonl --parallel 64```

image

## Speed Tests and Profiling

No influence, just bugfix

## Checklist

- [x] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [x] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [x] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review and Merge Process

1. Ping Merge Oncalls to start the process. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - Common commands include `/tag-and-rerun-ci`, `/tag-run-ci-label`, `/rerun-failed-ci`
4. After green CI and required approvals, ask Merge Oncalls or people with Write permission to merge the PR.

### CODE REVIEW

File: python/sglang/srt/layers/quantization/modelslim/modelslim.py [hunk 1]
```diff
@@ -163,57 +157,49 @@ def get_quant_method(
...
-        elif quant_type == "W4A4_DYNAMIC":
-            return ModelSlimW4A4Int4(
-                quant_config=self.quant_description, prefix=layer_name
-            )
-        raise NotImplementedError("No modelslim compatible scheme was found.")
-
     def get_linear_scheme(
-        self, layer: torch.nn.Module, layer_name: Optional[str] = None
+        self, layer: torch.nn.Module, prefix: Optional[str] = None
     ) -> Optional[ModelSlimLinearScheme]:
```
Comment (ping1jing2) [inline python/sglang/srt/layers/quantization/modelslim/modelslim.py:173]: why do you change `layer_name` into `prefix` here?

Comment (OrangeRedeng) [reply python/sglang/srt/layers/quantization/modelslim/modelslim.py:173]: Now we are having some confusion, somewhere the ```prefix``` is used, somewhere the ```layer_name```, I just brought it to one form. For example the same MoE function used ```prefix``` instead of ```layer_name``` 
image

Comment (OrangeRedeng) [reply python/sglang/srt/layers/quantization/modelslim/modelslim.py:173]: And previously we used ```lsayer_name``` as ```prefix``` in  ```_get_scheme_from_parts()```

Comment (OrangeRedeng) [reply python/sglang/srt/layers/quantization/modelslim/modelslim.py:173]: @TamirBaydasov What is your opinion?

Comment (TamirBaydasov) [reply python/sglang/srt/layers/quantization/modelslim/modelslim.py:173]: We never really work with full `layer_name` from config, it's always some kind of prefixes. Even in this function we pass our `prefix_in_quant_config` variable, not the whole layer name, so I am ok with this change, especially since `get_moe_scheme` uses `prefix` parameter as well. I overlooked it while implementing it the first time.
